src/app/settings/keyboard-mapping-settings/components/edit-keyboard-mapping/edit-keyboard-mapping.component.ts

```diff
@@ -97,4 +102,60 @@ export class EditKeyboardMappingComponent implements OnInit {
   public cancel(): void {
     this.onCancel.emit()
   }
+
+  public get formValidationTooltip(): string | undefined {
+    const invalidControls = this.findInvalidControls()
+    return invalidControls.length > 0 ? $localize`common.form-required-field.button-tooltip` + `: ${invalidControls.join(', ')}` : undefined
+  }
+
+  private findInvalidControls(): string[] {
+    const invalidControlLabels = []
+    const controls = { ...this.actionTriggerForm.controls, ...this.actionTriggerDataForm.controls }
+    for (const name in controls) {
+      if (controls[name].valid) {
+        continue
+      }
+      const translateControl = this.translateControl(name)
+      if (translateControl) {
+        invalidControlLabels.push(translateControl)
+      }
+    }
+    return invalidControlLabels
+  }
+
+  private translateControl(controlName: string): string | undefined {
+    switch (controlName) {
+      case FORM_TRIGGER_ON_CONTROL_LABEL:
+        return $localize`action-triggers.trigger-on.label`
+      case FORM_KEYS_CONTROL_LABEL:
+        return $localize`action-triggers.shortcut.label`
+      case FORM_ACTION_CONTROL_LABEL:
+        return $localize`action-triggers.action.label`
+      case FORM_ACTION_ARGUMENTS_CONTROL_LABEL:
+        return this.selectedAction?.argument?.name
+      default:
+        return undefined
+    }
+  }
+
+  public checkControlForErrors(name: string, groupName?: string): boolean {
+    const control = this.getFormControl(name, groupName)
+    if (!!groupName) {
+      return control ? control.invalid && control.touched : false
+    } else {
+      return control ? control.invalid && control.touched : false
+    }
+  }
+
+  private getFormControl(name: string, groupName?: string): AbstractControl | undefined | null {
```

Comment:
We don't want to return `null`. Ideally, we also don't want to return `undefined`, but that's not always an option.

Reply:
This is remove it after code changes